realm/realm-library/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -160,6 +164,51 @@ public DynamicRealmObject createObject(String className, Object primaryKeyValue)
                 CheckedRow.getFromRow(OsObject.createWithPrimaryKey(table, primaryKeyValue)));
     }
 
+    /**
+     * Instantiates and adds a new embedded object to the Realm.
+     * <p>
+     * This method should only be used to create objects of types marked as embedded.
+     *
+     * @param className the class name of the object to create.
+     * @param parentObject The parent object which should hold a reference to the embedded object.
+     *                     If the parent property is a list the embedded object will be added to the
+     *                     end of that list.
+     * @param parentProperty the property in the parent class which holds the reference.
+     * @return the newly created embedded object.
+     * @throws IllegalArgumentException if {@code clazz} is not an embedded class or if the property
+     * in the parent class cannot hold objects of the appropriate type.
+     * @see RealmClass#embedded()
+     */
+    public DynamicRealmObject createEmbeddedObject(String className,
+                                                   RealmModel parentObject,
```

Comment:
I think we we should use DynamicRealmObject here. By using RealmModel it opens up for people accidentially using their typed model classes